goal_src/jak2/engine/gfx/mood/mood.gc

```diff
@@ -5,16 +5,990 @@
+    (cond
+      ((nonzero? (-> gp-0 flicker-on))
+       (set! (-> arg0 times arg1 w) 1.0)
+       (if (not (paused?))
+           (+! (-> gp-0 flicker-on) -1)
+           )
+       )
+      ((nonzero? (-> gp-0 flicker-off))
+       (if (not (paused?))
+           (+! (-> gp-0 flicker-off) -1)
+           )
+       )
+      (else
+        (set! (-> gp-0 flicker-on) (the-as uint (the int (rand-vu-float-range 2.0 20.0))))
+        (if (zero? (the int (rand-vu-float-range 0.0 3.0)))
+            (set! (-> gp-0 flicker-off) (the-as uint (the int (rand-vu-float-range 2.0 120.0))))
+            (set! (-> gp-0 flicker-off) (the-as uint (the int (rand-vu-float-range 2.0 20.0))))
+            )
+        )
+      )
+    )
+  (none)
+  )
+
+(deftype florescent-state (structure)
+  ((value  float  :offset-assert   0)
+   (delay  int8   :offset-assert   4)
+   (delay2 int8   :offset-assert   5)
+   )
+  :method-count-assert 9
+  :size-assert         #x6
+  :flag-assert         #x900000006
+  )
+
+
+(defun update-mood-florescent ((arg0 mood-context) (arg1 int) (arg2 int))
+  "Unused"
+  (let ((gp-0 (the-as florescent-state (+ (+ arg2 1840) (the-as int arg0)))))
+    (set! (-> arg0 times arg1 w) (-> gp-0 value))
+    (when (not (paused?))
+      (cond
+        ((zero? (-> gp-0 delay))
+         (set! (-> gp-0 delay2) (the int (rand-vu-float-range 10.0 60.0)))
+         (set! (-> gp-0 delay) (the int (rand-vu-float-range 60.0 120.0)))
+         )
+        (else
+          (+! (-> gp-0 delay) -1)
+          )
+        )
+      (cond
+        ((>= (-> gp-0 delay2) (-> gp-0 delay))
+         (let ((f0-5 (rand-vu-float-range 1.0 1.5)))
+           (set! (-> gp-0 value) f0-5)
+           f0-5
+           )
+         )
+        ((< (-> gp-0 delay2) (-> gp-0 delay))
+         (let ((f0-6 1.5))
+           (set! (-> gp-0 value) f0-6)
+           f0-6
+           )
+         )
+        )
+      )
+    )
+  )
+
+(deftype electricity-state (structure)
+  ((value float  :offset-assert   0)
+   (scale float  :offset-assert   4)
+   )
+  :pack-me
+  :method-count-assert 9
+  :size-assert         #x8
+  :flag-assert         #x900000008
+  )
+
+
+;; WARN: Return type mismatch float vs none.
+(defun update-mood-electricity ((arg0 mood-context) (arg1 int) (arg2 int) (arg3 float) (arg4 float))
+  (let ((gp-0 (the-as electricity-state (+ (+ arg2 1840) (the-as int arg0)))))
+    (set! (-> arg0 times arg1 w) (* (-> gp-0 value) (-> gp-0 scale)))
+    (if (not (paused?))
+        (set! (-> gp-0 value) (rand-vu-float-range arg3 arg4))
+        )
+    )
+  (none)
+  )
+
+(deftype pulse-state (structure)
+  ((pulse float  :offset-assert   0)
+   )
+  :pack-me
+  :method-count-assert 9
+  :size-assert         #x4
+  :flag-assert         #x900000004
+  )
+
+
+;; WARN: Return type mismatch float vs none.
+(defun update-mood-pulse ((arg0 mood-context) (arg1 int) (arg2 int) (arg3 float) (arg4 float) (arg5 float) (arg6 float))
```

Comment:
I wonder what `arg2` in these functions actually is

Reply:
we'll find out eventually!